Modules/Sources/DesignSystem/Components/FAB.swift

```diff
@@ -0,0 +1,25 @@
+import SwiftUI
+
+public struct FAB: View {
+    let image: Image
+    let action: () -> Void
+
+    @Environment(\.colorScheme) var colorScheme
+
+    public init(image: Image = Image(systemName: "plus"), action: @escaping () -> Void) {
+        self.image = image
+        self.action = action
+    }
+
+    public var body: some View {
+        Button(action: action) {
+            image
+                .font(.title2)
+                .foregroundStyle(Color.white)
+                .frame(width: 50, height: 50)
```

Comment:
Should this scale with the dynamic type size?

Reply:
Good catch. I added the following:

```swift
@ScaledMetric(relativeTo: .title2) private var size = 50.0
```

Looks great now:

<img width="1349" alt="Screenshot 2024-07-22 at 8 07 18 PM" src="https://github.com/user-attachments/assets/3f73b9fe-3da8-487b-bf3c-9753cbd9a078">

Reply:
Hmm, actually we should probably stop at around AX1. The other variants are too large and will start covering the content.

```swift
.dynamicTypeSize(...DynamicTypeSize.accessibility1)
```

<img width="1229" alt="Screenshot 2024-07-23 at 7 30 51 AM" src="https://github.com/user-attachments/assets/628ffe89-3e16-4087-a965-cc476625deb7">